README.md

```diff
@@ -8,59 +8,172 @@ The backend for a generic mobile-friendly quiz engine created using FastAPI and
 
 ## Installation
 
+### Local DB Setup
+
+- The following steps are for a Mac.
+- To run the backend locally, you would need to setup a local instance of mongodb. The offical instructions [here](https://www.mongodb.com/docs/manual/tutorial/install-mongodb-on-os-x/#run-mongodb-community-edition) are good and simple enough to follow. Those steps are also listed down below.
+
+  - Install the Xcode command-line tools
+    ```bash
+    xcode-select --install
+    ```
+  - Make sure you have homebrew installed. If not, go [here](https://brew.sh/#install)
+  - Download the official Homebrew formula for MongoDB
+    ```bash
+    brew tap mongodb/brew
+    ```
+  - Update Homebrew and all existing formulae
+    ```bash
+    brew update
+    ```
+  - Install MongoDB
+    ```bash
+    brew install mongodb-community@6.0
+    ```
+
+### Virtual Environment Setup
+
 - Create a virtual environment (make sure that `virtualenv` is installed on your system):
+  ```bash
+  virtualenv venv
+  ```
+
+- Activate the environment:
+  ```bash
+  source venv/bin/activate
+  ```
+
+- Install the dependencies:
+  ```bash
+  pip install -r app/requirements.txt
+  ```
+
+- Install `pre-commit`:
+  ```
+  pip install pre-commit
```

Comment:
Why can't `pre-commit` be included in `requirements.txt` . Will save this one command

Reply:
I didn't change this part much, it was there since before.
I think the reasoning is that we don't wanna push the pre-commit package in the build that will be deployed to lambda as we don't need it at runtime

Reply:
Ah okay. Seems overkill right now to have separate `requirements-dev` and `requirements` .. so cool cool